src/hotspot/os/linux/os_linux.cpp

```diff
@@ -423,7 +423,7 @@ static const char *unstable_chroot_error = "/proc file system not found.\n"
                      "environment on Linux when /proc filesystem is not mounted.";
 
 void os::Linux::initialize_system_info() {
-  set_processor_count(sysconf(_SC_NPROCESSORS_CONF));
+  set_processor_count(checked_cast<int>(sysconf(_SC_NPROCESSORS_CONF)));
```

Comment:
What are the costs of checked_cast?

Reply:
checked_cast instantiates a template or two. I haven't measured code bloat.  It carries risk of asserting for odd inputs that used to be truncated.  In this case, it seems like a good place to have the assert.  In obvious places, it seems like noise.

Reply:
There are a few ways to fix code like this without changing the call site:
1. overflow func() to take the alternative type, and do the checked_cast there
2. change func() to a template function
3. Change func() to take a user-defined type whose constructor accepts both types
I've experimented with all 3 in the compiler code, but I don't know what others think of tricks like these.
I think 2) is already used it HotSpot.

Reply:
@coleenp: The only case where sysconf could be not a small positive integer is when it returns an error, -1, and checked_cast will not catch that since it still fits an int. 

Just my opinion, but I'd either leave this unchecked, or convert the parameter for set_processor_count to unsigned.

@dean-long:
> change func() to a template function

I hope we don't do that, that sounds terrible. I still have to find a modern IDE that can work well with all the template code we have. The more templates we use, the worse features like call graph display work in IDEs like CDT or CLion.

Reply:
I'm also not in favor of additional templates to hide our use of wrongly sized integers.  There's already enough time-consuming decoding in places and terrible error messages.  I'd rather the integers be the right size.
I don't like tricks. I'm not opposed at all to direct C style casts where we know they fit.  Seems simplest.  I'll change this one.

Reply:
I think checked_cast should be used when the variable is potentially unconstrained, but as Thomas confirms these sysconf calls are only ever going to return small integers.